in_toto/models/link.py

```diff
@@ -181,3 +181,9 @@ def _validate_environment(self):
       raise securesystemslib.exceptions.FormatError(
           "Invalid Link: field `environment` must be of type dict, got: {}"
           .format(type(self.environment)))
+    for key in self.environment.keys():
```

Comment:
We should remove this, the specification doesn't mandate a format for `environment` and `byproducts`. https://github.com/in-toto/docs/blob/master/in-toto-spec.md#441-environment-information